lib/algos/heap/Heap.ts

```diff
@@ -0,0 +1,124 @@
+export enum HeapOrder {
+    Min = -1,
+    Max = 1,
+}
+
+export enum CompareResult {
+    LT = -1,
+    EQ = 0,
+    GT = 1,
+}
+
+export type CompareFunction = (x: any, y: any) => CompareResult;
+
+export class Heap {
+    size: number;
+    _maxSize: number;
+    _order: HeapOrder;
+    _heap: any[];
+    _cmpFn: CompareFunction;
```

Comment:
Do we want to enforce private members?

```suggestion
    #maxSize: number;
    #order: HeapOrder;
    #heap: any[];
    #cmpFn: CompareFunction;
```